### PR TITLE
[Core] Implement object usefull for string formatting

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Related tickets |
| License         | MIT

There are many places where we need to format string to some *code-like* form and it is nice to have an object with such a responsibility. I encourage you all to talk about this class name. I've had many ideas like:
- ``SpacesToUnderscoreFormatter``
- ``SnakeCaseFormatter``
- ``Snaker`` (:smile:)

but finally I decided to something more neutral. I would appreciate brainstorm about that ;)

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
```StringInflector::nameTo(Lowercase|Uppercase)Code```

https://www.google.pl/#q=inflect

WDYT?

**Comment by Zales0123** [comment]
Looks nice 👍  I'm only not convinced with ``Code`` at the end of function, what if we want to use this ``Inflector`` not for code but sth else? It could lead to confusion and a huge disaster :(

**Comment by pjedrzejewski** [comment]
@Zales0123 I believe separation of concerns is good, so if we define behavior of this object as transforming resource names to codes, if anything about that behavior changes, we change it here. Converting spaces to underscores is generic string manipulation functionality and could go to other object if we need it, but let's think about it when we actually need it. In Behat's case, we always (I think) need to transform name to code.

**Comment by Zales0123** [comment]
@pjedrzejewski convinced and agreed 👍

**Comment by michalmarcinkowski** [comment]
Since the object is in `Core` component we should not use it in bundles that are not dependent on `Core`.  Please remove the usages from `ProductContext`, `PromotionContext` etc. they will be removed soon anyway.

File: src/Sylius/Component/Core/Formatter/StringInflector.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        return str_replace(' ', '_', $value);
+    }
+
+    /**
+     * @param string $value
+     *
+     * @return string
+     */
+    public static function nameToLowercaseCode($value)
```
Comment (lchrusciel) [inline src/Sylius/Component/Core/Formatter/StringInflector.php]: nameToLowercase**d**Code? WDYT?

Comment (TheMadeleine) [inline src/Sylius/Component/Core/Formatter/StringInflector.php]: Nope :D lowercase is enough

**Comment by pjedrzejewski** [comment]
Thank you Mateusz!

File: src/Sylius/Behat/Context/Setup/ProductContext.php [hunk 2]
```diff
@@ -428,6 +429,6 @@ private function saveProduct(ProductInterface $product)
      */
     private function convertToCode($productName)
     {
-        return strtoupper(str_replace(' ', '_', $productName));
+        return StringInflector::nameToUpercaseCode($productName);
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Setup/ProductContext.php]: Why we use uppercase for products code?

**Pull request merged**